File: day1.py
from common import *

ip = aoc_input("1")
ls = lines(ip)

def part1():
    ints = [[int(c) for c in l if c.isdigit()] for l in ls]
    print(sum(i[0] * 10 + i[-1] for i in ints))

def part2():
    import re
    textdigits = ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    rmap = {x: i + 1 for i, x in enumerate(textdigits)}

    acc = 0
    for l in ls:
        start = re.match("^.*?(one|two|three|four|five|six|seven|eight|nine|\d).*$", l)[1]
        end = re.match("^.*(one|two|three|four|five|six|seven|eight|nine|\d).*?$", l)[1]

        if start in rmap:
            start = rmap[start]
        else:
            start = int(start)

        if end in rmap:
            end = rmap[end]
        else:
            end = int(end)

        acc += start * 10 + end

    print(acc)

part2()

File: day1.jl
include("common.jl")

ls = aoclines("1")

function part1()
  ns = [filter(isdigit, l) for l in ls]
  ss = [int(x[1]) * 10 + int(x[end]) for x in ns]
  pln(sum(ss)) 
end

function part2()
  numbers = split("one two three four five six seven eight nine", " ")
  rmap = Dict(x => i for (i, x) in enumerate(numbers))

  matchtonum(m) = if isdigit(m[1]) int(m) else rmap[m] end
  
  function parseline(l)
    front = match(r"^.*?(one|two|three|four|five|six|seven|eight|nine|\d).*$", l)[1]
    back = match(r"^.*(one|two|three|four|five|six|seven|eight|nine|\d).*?$", l)[1]
    
    return matchtonum(front) * 10 + matchtonum(back)
  end

  pln(sum(parseline(l) for l in ls))
end

part2()
